lib/devfs/stdout.c

```diff
@@ -44,9 +44,45 @@
 #ifdef CONFIG_LIBDEVFS_DEV_STDOUT
 #define DEV_STDOUT_NAME "stdout"
 
+/* TODO: Some consoles require both a newline and a carriage return to
```

Comment:
We plan to add a TTY layer, right? IMO the TTY layer is where this replicated code should go. IIRC I discussed this with @skuenzer and replicating the code is what we settled on for now. Of course, we can decide to move it to a common place now given how often this code is replicated. I think when we touched `uk_scanf` in the `ukconsole` PR, we also decided that it should eventually go into the TTY layer. So that's two items for the TTY layer already

Reply:
Yes, longer term the place needs to be posix-tty and maybe nolibc for compatibility mode when we build without `posix-tty`/`posix-fdtab`. So I thinnk the duplication for now is okay given that we did not finished architecting TTY handling correctly yet.